website_sale_product_search_advanced/models/product_template.py

```diff
@@ -0,0 +1,19 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import models, fields, api
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    website_partner_ref = fields.Char(
+        string='Partner Reference'
+    )
+
+    @api.model
+    def _search_get_detail(self, website, order, options):
+        res = super()._search_get_detail(website, order, options)
+        res['search_fields'].append('barcode')
```

Comment:
Locally tested, when any result is obtained seems to be :+1: , but when no results were found, in internal error is raised looking for similar results:

![imagen](https://github.com/solvosci/slv-e-commerce/assets/12749832/19967d06-af48-4a67-a32e-23604f6dd735)

You can try this very interesting "similar results" functionality searching with a not exact word like "florr" instead of "floor", with this addon uninstalled.

Error is raised in those cases because Odoo looks via Postgres SQL raw query for a similar result... and barcode is actually a field of product.product model, not template (see https://github.com/odoo/odoo/blob/b6b90636938ae961c339807ea893cabdede9f549/addons/product/models/product_template.py#L130 and https://github.com/odoo/odoo/blob/b6b90636938ae961c339807ea893cabdede9f549/addons/product/models/product_product.py#L44-L46). This fixes it:

```suggestion
        res['search_fields'].append('product_variant_ids.barcode')
```